internal/driver/readhandler.go

```diff
@@ -24,23 +46,86 @@ func (d *Driver) HandleReadCommands(deviceName string, protocols map[string]mode
 
 	d.Logger.Debugf("Driver.HandleReadCommands: protocols: %v resource: %v attributes: %v", protocols, reqs[0].DeviceResourceName, reqs[0].Attributes)
 
-	// create device client and open connection
-	endpoint, err := config.FetchEndpoint(protocols)
+	endpoints, err := opcua.GetEndpoints(d.serviceConfig.OPCUAServer.Endpoint)
```

Comment:
This is not a review but rather something I personally like myself, maybe you will like it as well. 
I like to have descriptive variables, like when we read them we know exactly what they are. Something like serverEndpoints is what I would have chosen here. But, like I said my personal preference.

Reply:
true, done